modules/apps/captcha/captcha-taglib/src/main/resources/META-INF/resources/captcha/simplecaptcha.jsp

```diff
@@ -65,23 +68,44 @@ String url = (String)request.getAttribute("liferay-captcha:captcha:url");
 
 			if (refreshCaptcha && !hasEventAttached) {
 				hasEventAttached = true;
-				refreshCaptcha.addEventListener('click', () => {
-					var url = Liferay.Util.addParams(
-						't=' + Date.now(),
-						'<%= HtmlUtil.escapeJS(url) %>'
-					);
-
-					var captcha = document.getElementById(
-						'<portlet:namespace />captcha'
-					);
-
-					if (captcha) {
-						captcha.setAttribute('src', url);
-					}
-				});
+				refreshCaptcha.addEventListener('click', handleCaptcha);
 			}
 		}
 
+		function handleCaptcha() {
+			if (<%= url.startsWith("/o/") %>) {
+				return getCaptchaSourceImage('<%= url %>');
+			}
+
+			let url = Liferay.Util.addParams(
+				't=' + Date.now(),
+				'<%= HtmlUtil.escapeJS(url) %>'
+			);
+
+			captcha.setAttribute('src', url);
+		}
+
+		function getCaptchaSourceImage(url) {
+			Liferay.Util.fetch(url, {
+				method: 'GET',
+			})
+			.then((response) => {
+				if (!response.ok) {
+					throw new Error();
+				}
+
+				return response.json();
+			})
+			.then((response) => {
+				captcha.setAttribute('src', response.image);
+			})
+			.catch((error) => {
+				console.error('An error occurred:', error);
```

Comment:
please not log the error, if you want to log the error, firstly, you should check the portal which is in the development status

Reply:
Fixed it. Thanks